torch/cuda/__init__.py

```diff
@@ -485,6 +486,8 @@ def _strtoul(s: str) -> int:
     return rc
 
 def _raw_device_count_nvml() -> int:
+    """Return number of devices as reported by NVML
+    or negative value if NVML discovery/initialization failed."""
     from ctypes import CDLL, c_int
     nvml_h = CDLL("libnvidia-ml.so.1")
```

Comment:
also, I wonder if this dependency on nvml is stable across cuda releases? should the library file name be hardcoded in some other single special file where some other dependencies are listed? (I don't know if pytorch has such a place) just wonder that at some point nvidia will upgrade it to .so.2 and pytorch would forget to update this, or how it would operate when both .so.1 and .so.2 are in the wild and both need to be supported

Reply:
nvml is not part of the CUDA, but rather a driver. And yes, it appears to be quite stable, i.e. there were always `libcuda.so.1` and `libnvml.so.1` in all CUDA releases since 6.0 (or may be even older)